server: TestClusterVersionMixedVersionTooOld failed

[(server).TestClusterVersionMixedVersionTooOld failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2444785&tab=buildLog) on [master@5d8f1d3dbc356573a1fb2d1d6bf564bf0370daff](https://github.com/cockroachdb/cockroach/commits/5d8f1d3dbc356573a1fb2d1d6bf564bf0370daff):

```
=== RUN   TestClusterVersionMixedVersionTooOld
    test_log_scope.go:155: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestClusterVersionMixedVersionTooOld355292341
    test_log_scope.go:64: use -show-logs to present logs inline
    version_cluster_test.go:438: 0 pq: n4 required, but unavailable
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestClusterVersionMixedVersionTooOld355292341
--- FAIL: TestClusterVersionMixedVersionTooOld (0.43s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestClusterVersionMixedVersionTooOld PKG=./pkg/server TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestClusterVersionMixedVersionTooOld.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

[(server).TestClusterVersionMixedVersionTooOld failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2457792&tab=buildLog) on [master@8ff64f5d897a9c7b39ca437193a11a7300cecbc1](https://github.com/cockroachdb/cockroach/commits/8ff64f5d897a9c7b39ca437193a11a7300cecbc1):

```
=== RUN   TestClusterVersionMixedVersionTooOld
    test_log_scope.go:155: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestClusterVersionMixedVersionTooOld707123688
    test_log_scope.go:64: use -show-logs to present logs inline
    version_cluster_test.go:439: 0 pq: n4 required, but unavailable
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestClusterVersionMixedVersionTooOld707123688
--- FAIL: TestClusterVersionMixedVersionTooOld (0.46s)

```

<details><summary>More</summary><p>
Parameters:

- TAGS=
- GOFLAGS=-parallel=4

```
make stressrace TESTS=TestClusterVersionMixedVersionTooOld PKG=./pkg/server TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```


[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2ATestClusterVersionMixedVersionTooOld.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

--- a/pkg/server/version_cluster_test.go
+++ b/pkg/server/version_cluster_test.go
@@ -433,11 +433,23 @@ func TestClusterVersionMixedVersionTooOld(t *testing.T) {
 
 	// The other nodes are just as careful.
 	for i := 0; i < len(versions)-2; i++ {
-		if err := tc.setVersion(i, v1s); !testutils.IsError(err,
-			fmt.Sprintf("binary version %s less than target version %s", v0s, v1s),
-		) {
-			t.Fatal(i, err)
-		}
+		testutils.SucceedsSoon(t, func() error {
+			err := tc.setVersion(i, v1s)
+			if testutils.IsError(err, "required, but unavailable") {
+				// Paper over transient unavailability errors. Because we're
+				// setting the cluster version so soon after cluster startup,
+				// it's possible that we're doing so before all the nodes have
+				// had a chance to heartbeat their liveness records for the very
+				// first time. To other nodes it appears that the node in
+				// question is unavailable.
+				return err
+			}
+
+			if !testutils.IsError(err, fmt.Sprintf("binary version %s less than target version %s", v0s, v1s)) {
+				t.Fatal(i, err)
+			}
+			return nil
+		})
 	}
 
 	// Check that we can still talk to the first three nodes.

